app/editor/src/features/admin/ingests/IngestList.tsx

```diff
@@ -16,21 +25,98 @@ const IngestList: React.FC<IIngestListProps> = (props) => {
   const navigate = useNavigate();
   const [{ ingests }, { findAllIngests, setIngestEnabledStatus }] = useIngests();
   const [{ ingestFilter }] = useAdminStore();
-
-  const items = ingests.filter(
-    (i) =>
-      i.name.toLocaleLowerCase().includes(ingestFilter) ||
-      i.source?.code.toLocaleLowerCase().includes(ingestFilter) ||
-      i.description.toLocaleLowerCase().includes(ingestFilter) ||
-      i.ingestType?.name.toLocaleLowerCase().includes(ingestFilter) ||
-      getStatus(i).toLocaleLowerCase().includes(ingestFilter),
-  );
+  const [isReady, setIsReady] = React.useState(false);
+  const [items, setItems] = React.useState<IIngestModel[]>([]);
 
   React.useEffect(() => {
-    findAllIngests().catch(() => {});
-    // Only init list.
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+    if (!ingests.length && !isReady) {
+      setIsReady(true);
+      findAllIngests()
+        .then((data) => {
+          setItems(
+            data.filter(
+              (i) =>
+                i.name.toLocaleLowerCase().includes(ingestFilter) ||
+                i.source?.code.toLocaleLowerCase().includes(ingestFilter) ||
+                i.description.toLocaleLowerCase().includes(ingestFilter) ||
+                i.ingestType?.name.toLocaleLowerCase().includes(ingestFilter) ||
+                getStatus(i).toLocaleLowerCase().includes(ingestFilter),
+            ),
+          );
+        })
+        .catch(() => {});
+    } else {
+      setItems(
+        ingests.filter(
+          (i) =>
+            i.name.toLocaleLowerCase().includes(ingestFilter) ||
+            i.source?.code.toLocaleLowerCase().includes(ingestFilter) ||
+            i.description.toLocaleLowerCase().includes(ingestFilter) ||
+            i.ingestType?.name.toLocaleLowerCase().includes(ingestFilter) ||
+            getStatus(i).toLocaleLowerCase().includes(ingestFilter),
+        ),
+      );
+    }
+  }, [findAllIngests, isReady, ingests, ingestFilter]);
+
+  const handleOnSorting = (column: any, direction: any) => {
+    if (direction && column.name) {
+      setItems((items) => {
+        switch (column.name) {
+          case 'source.code':
+            return [...items].sort((a: any, b: any) => {
```

Comment:
we could probably centralize this repeated code to a single function to make it easier to maintain.